### PR TITLE
feat(JOML): migrate world generation

### PR DESCRIPTION
**Pull request by jdrueckert**

### CODE REVIEW

File: src/test/java/org/terasology/cities/bldg/DebugRasterTarget.java [hunk 1]
```diff
@@ -16,38 +16,39 @@
...
-    private final Region3i region;
+    private final BlockRegion region;
     private final TeraArray data;
     private final List<BlockType> mapping = new ArrayList<BlockType>();
 
     public DebugRasterTarget(int min, int max) {
         this.data = new TeraDenseArray16Bit(SIZE_X, max - min + 1, SIZE_Z);
         this.area = Rect2i.createFromMinAndMax(0, 0, SIZE_X, SIZE_Z);
-        this.region = Region3i.createFromMinMax(new Vector3i(0, min, 0), new Vector3i(SIZE_X, max, SIZE_Z));
+        this.region = BlockRegions.createFromMinAndMax(new Vector3i(0, min, 0), new Vector3i(SIZE_X, max, SIZE_Z));
```
Comment (skaldarnar) [inline src/test/java/org/terasology/cities/bldg/DebugRasterTarget.java]: This no longer exists
```suggestion
        this.region = new BlockRegion(0, min, 0, SIZE_X, max, SIZE_Z);
```

File: src/test/java/org/terasology/cities/bldg/DebugRasterTarget.java [hunk 2]
```diff
@@ -58,7 +59,7 @@ public void setBlock(int x, int y, int z, BlockType type) {
             index = mapping.size();
             mapping.add(type);
         }
-        data.set(x, y - region.minY(), z, index);
+        data.set(x, y - region.getMinY(), z, index);
```
Comment (skaldarnar) [inline src/test/java/org/terasology/cities/bldg/DebugRasterTarget.java]:
```suggestion
        data.set(x, y - region.minY(), z, index);
```

**Review by skaldarnar — CHANGES_REQUESTED** [review]
Maybe this PR was not updated after the latest changes?

File: src/test/java/org/terasology/cities/bldg/DebugRasterTarget.java [hunk 3]
```diff
@@ -87,7 +88,7 @@ public BlockType get(int index) {
 
             @Override
             public int size() {
-                return region.maxY() - region.minY() + 1;
+                return region.getMaxY() - region.getMinY() + 1;
```
Comment (jdrueckert) [inline src/test/java/org/terasology/cities/bldg/DebugRasterTarget.java]:
```suggestion
                return region.maxY() - region.minY() + 1;
```

**Review by skaldarnar — DISMISSED** [review]

**Review by skaldarnar — APPROVED** [review]

**Pull request merged**